build/gulp.migration.js

```diff
@@ -0,0 +1,133 @@
+require('bluebird-global')
+const path = require('path')
+const fs = require('fs')
+const mkdirp = require('mkdirp')
+const exec = require('child_process').exec
+const fse = require('fs-extra')
+const glob = require('glob')
+const tmp = require('tmp')
+const dotenv = require('dotenv')
+
+const dumpServerData = async () => {
+  const args = require('yargs')(process.argv).argv
+
+  const rootPath = args.path || './'
+  const archiveName = args.name || 'test-migration'
+  const pgBinPath = args.pgPath || __dirname
+
+  const tmpDir = tmp.dirSync({ unsafeCleanup: true })
+
+  const dataFolder = await getValidPath(['out/bp/data', 'data'], rootPath)
+  const dataItems = await fse.readdir(dataFolder)
+
+  for (const item of dataItems.filter(x => x !== 'assets')) {
+    await fse.copy(path.resolve(dataFolder, item), `${tmpDir.name}/${item}`)
+  }
+
+  try {
+    mkdirp.sync(`${tmpDir.name}/storage`)
+    await dumpPostgresData(`${tmpDir.name}/storage`, rootPath, pgBinPath)
+  } catch (err) {
+    console.error('Error:', err)
+
+    console.error(`Could not dump the postgres database.
+Postgres must be installed, and the tool pg_dump must be available on your PATH (use --pgPath to override it)
+Please make sure that the credentials provided in DATABASE_URL are valid`)
+    throw err
+  }
+
+  const files = await Promise.fromCallback(cb => glob('**/*', { cwd: tmpDir.name, nodir: true, dot: true }, cb))
+
+  // Not requiring the file at the top of the file, because it may not have been built yet
+  const { createArchive } = require(path.resolve('./out/bp/core/misc/archive'))
+
+  const currentVersion = await getVersion(rootPath)
+  const filename = await createArchive(`./${archiveName}_${currentVersion}.tgz`, tmpDir.name, files)
+
+  console.log(`Complete server archive saved at ${path.join(__dirname, filename)}.`)
+  tmpDir.removeCallback()
```

Comment:
To make sure we don't "pollute" the tmp directory of someone (when something fails), I think it would be better to call the remove callback inside the finally statement of a try catch. I know a server restart would clear the folder, but still I think I would be a best practice. What do you think?

Reply:
That's when a developer creates a dump of an installation, the temp folder should be removed as soon as the gulp process is done, which is like imemdiately